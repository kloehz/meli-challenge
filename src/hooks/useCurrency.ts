import { getCurrency } from "../constants/currency";
import { Price } from "../types/getItems";


const useCurrency = ({currency, amount, decimals}: Price) => {
    const currencyType = getCurrency(currency);
    let newAmount: string;
    newAmount = new Intl.NumberFormat().format(Number(amount)).toString();
    const decimalsString = decimals === 0 ? `${decimals.toString()}0` : decimals.toString();

    return {currency: currencyType, amount, decimals: decimalsString};
}

export default useCurrency;