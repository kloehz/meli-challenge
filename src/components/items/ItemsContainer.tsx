import useCurrency from '../../hooks/useCurrency';
import { Item } from '../../types/getItems';
import { useHistory } from 'react-router-dom';
import { Prices } from '../prices/Prices';

export const ItemsContainer = ({id, picture, title, price, seller_address}: Item) => {

    const history = useHistory();

    const handleClick = () => {
            history.push({pathname: `/items/${id}`});
    };

    const parsedPrice = useCurrency({...price});

    return (
        <div className="product-container" onClick={handleClick}>
            <div className="item-image-container">
                <img src={picture} />
            </div>
        <div className="title-container">
            <Prices price={parsedPrice} style='item' />
                <h4>{title}</h4>
            </div>
            <div className="seller-address-container">
                <h4 className="seller-address">{seller_address}</h4>
            </div>
        </div>
    )
}
