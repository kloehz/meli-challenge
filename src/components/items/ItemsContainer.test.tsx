import { shallow } from "enzyme"
import useCurrency from "../../hooks/useCurrency";
import { ItemsContainer } from "./ItemsContainer"

const mockHistoryPush = jest.fn();
const mockUseCurrency = jest.fn();

jest.mock('react-router', () => ({
    // @ts-ignore
    ...jest.requireActual('react-router'),
    useHistory: () => ({
      push: mockHistoryPush,
    }),
}));



const itemMock = {
    "id":"MLA877200255",
    "picture":"http://http2.mlstatic.com/D_923689-MLA43399324060_092020-O.jpg",
    "title":"Vaso Pampa Durax De Vidrio 400ml Agua Jugo X 12 Unid Oferta",
    "price": {
        "currency":"ARS",
        "amount":869,
        "decimals":0
    },
    "seller_address":"Capital Federal",
    "condition": "Nuevo",
    "free_shiping": true
}

describe('ItemsContainer tests', () => {
    
    const wrapper = shallow(<ItemsContainer {...itemMock} />);

    test('should be match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should be redirect on click', () => {
        const productContainer = wrapper.find('div').at(0);
        productContainer.simulate('click');
        expect(mockHistoryPush).toHaveBeenCalledWith({"pathname": '/items/MLA877200255'});
    });
    
    test('should be have the correct image', () => {
        expect(wrapper.find('img').prop('src')).toBe('http://http2.mlstatic.com/D_923689-MLA43399324060_092020-O.jpg');
    });
    
})
