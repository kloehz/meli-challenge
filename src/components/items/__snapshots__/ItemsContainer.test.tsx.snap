// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsContainer tests should be match snapshot 1`] = `ShallowWrapper {}`;
